File: Models/DomainModels/YoutubeChannel.cs
namespace Models.DomainModels;

public class YoutubeChannel : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Handle { get; set; } = string.Empty;
    public string ThumbnailUrl { get; set; } = string.Empty;
    public string BannerUrl { get; set; } = string.Empty;
    public int SubscriberCount { get; set; }
    public int VideoCount { get; set; }
    public string ChannelUrl => $"https://www.youtube.com/channel/{Id}";
    public string Avatar => $"/api/Thumbnail/channel/{Id}";
    public string Banner => $"/api/Thumbnail/channel/{Id}";
    public string Description { get; set; } = string.Empty;

    public ICollection<YoutubeVideo>? Videos { get; set; }
}

File: Ui/src/models/YoutubeChannel.ts
import { Model } from 'pinia-orm'
import YoutubeVideo from './YoutubeVideo'
import { ArrayCast } from 'pinia-orm/dist/casts'

export default class YoutubeChannel extends Model {
  static entity = 'channels'
  static piniaOptions = {
    persist: true
  }

  static fields() {
    return {
      id: this.attr(null),
      title: this.string(''),
      handle: this.string(''),
      thumbnailUrl: this.string(''),
      bannerUrl: this.string(''),
      subscriberCount: this.number(0),
      videoCount: this.number(0),
      channelUrl: this.string(''),
      avatar: this.string(''),
      banner: this.string(''),
      description: this.string(''),
      videos: this.hasMany(YoutubeVideo, 'youtubeChannelId', 'id')
    }
  }
    static casts() {
      return {
        videos: ArrayCast
      }
    }
}
